SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java

```diff
@@ -350,11 +355,16 @@ private void drainRequestCleanupQueue() {
               createLbCleanupRequest(requestId, matchingActiveTaskIds);
             }
           } else {
-            Optional<SingularityRequestHistory> maybeHistory = requestHistoryHelper.getLastHistory(requestId);
-            if (maybeHistory.isPresent() && maybeHistory.get().getRequest().isLoadBalanced() && configuration.isDeleteRemovedRequestsFromLoadBalancer()) {
-              createLbCleanupRequest(requestId, matchingActiveTaskIds);
+            if (matchingActiveTaskIds.iterator().hasNext()) {
```

Comment:
Maybe `Iterables.isEmpty(matchingActiveTaskIds)` just for cleanliness/readability